src/app/shared/guards/confirm-user.guard.ts

```diff
@@ -43,10 +45,9 @@ export class ConfirmUserGuard implements CanActivate {
         .subscribe((tokenObj) => {
             if (tokenObj) {
                 this.tokenService.storeToken(tokenObj);
-                if(localStorage.getItem('privateConsultationId')){
-                    let id = +localStorage.getItem('privateConsultationId');
-                    localStorage.setItem('privateConsultationId', '');
-                    this.router.navigate(['/consultations', id ]);
+                this.getCurrentUser();
+                if(this.currentUser.callbackUrl){
+                    this.router.navigateByUrl(this.currentUser.callbackUrl);
```

Comment:
@Harshwardhan18 This is not correct bro, 
this.getCurrentUser() function will be called, and after that before even this.currentUser is set, you are checking this.currentUser. You'll need to subscribe  `userLoaded$` in this function and in it's success callback, use currentUser.
This has happened previously as well, I'll suggest you to read more about observables, promises and asynchronousity

Reply:
@Manzurkds bro i didn't code it. it was written before and is working just fine.

Reply:
@Harshwardhan18 I don't understand how it's not your code bro, the particular thing I point was added in this pr.  This might work fine for you but looking at the code I don't think that'll always be the case. You might be lucky to have currentUser getting loaded somehow before the next line, but in theory, that's not correct.

Reply:
@Manzurkds my bad.